src/actions/settings.js

```diff
@@ -0,0 +1,64 @@
+import { toastr } from 'react-redux-toastr';
+import { GET_LANGUAGE_CHANNEL, SET_LANGUAGE_CHANNEL } from '../config/channels';
+import { ERROR_GENERAL } from '../config/errors';
+import {
+  ERROR_MESSAGE_HEADER,
+  ERROR_GETTING_LANGUAGE,
+  ERROR_SETTING_LANGUAGE,
+} from '../config/messages';
+import {
+  FLAG_GETTING_LANGUAGE,
+  FLAG_SETTING_LANGUAGE,
+  GET_LANGUAGE_SUCCESS,
+  SET_LANGUAGE_SUCCESS,
+} from '../types';
+import { createFlag } from './common';
+
+const flagGettingLanguage = createFlag(FLAG_GETTING_LANGUAGE);
+const flagSettingLanguage = createFlag(FLAG_SETTING_LANGUAGE);
+
+const getLanguage = async () => (dispatch) => {
+  try {
+    dispatch(flagGettingLanguage(true));
+    window.ipcRenderer.send(GET_LANGUAGE_CHANNEL);
+    window.ipcRenderer.once(GET_LANGUAGE_CHANNEL, (event, lang) => {
+      if (lang === ERROR_GENERAL) {
+        toastr.error(ERROR_MESSAGE_HEADER, ERROR_GETTING_LANGUAGE);
+      } else {
+        dispatch({
+          type: GET_LANGUAGE_SUCCESS,
+          payload: lang,
+        });
+      }
+      dispatch(flagGettingLanguage(false));
+    });
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.error(e);
+    toastr.error(ERROR_MESSAGE_HEADER, ERROR_GETTING_LANGUAGE);
```

Comment:
You'll probably need `dispatch(flagGettingLanguage(false));` in the `catch` block as well. (same below)